src/gui/gui_main_menu.cpp

```diff
@@ -113,6 +113,24 @@ void autosave_display::on_update(sys::state& state) noexcept {
 			break;
 	}
 }
+
+
+void tooltip_mode_left::button_action(sys::state& state) noexcept {
+	state.user_settings.bind_tooltip_mouse = !state.user_settings.bind_tooltip_mouse;
+}
+void tooltip_mode_left::on_update(sys::state& state) noexcept {
+	send(state, parent, notify_setting_update{});
```

Comment:
this needs to go into the button action, not on update

Reply:
You send the message up after changing something when the button is clicked to notify the main window that it needs to save and update. If you do it in on update, you will create a loop where, upadate -> notify parent to update -> parent updates again -> etc